applications/FluidDynamicsApplication/python_scripts/adjoint_fluid_solver.py

```diff
@@ -18,13 +18,25 @@ def __init__(self, model, settings):
         self.min_buffer_size = 2
 
     def AddDofs(self):
-        KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_VECTOR_1_X, self.main_model_part)
-        KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_VECTOR_1_Y, self.main_model_part)
-        KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_VECTOR_1_Z, self.main_model_part)
-        KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_SCALAR_1, self.main_model_part)
+        dofs_to_add = []
+        dofs_to_add.append("ADJOINT_FLUID_VECTOR_1_X")
+        dofs_to_add.append("ADJOINT_FLUID_VECTOR_1_Y")
+        dofs_to_add.append("ADJOINT_FLUID_VECTOR_1_Z")
+        dofs_to_add.append("ADJOINT_FLUID_SCALAR_1")
+        KratosMultiphysics.VariableUtils.AddDofsList(dofs_to_add, self.main_model_part)
```

Comment:
Why can't we use the `GetDofsList` to add the dofs. If so, it will make this consistent. Do we need the `ADJOINT_FLUID_VECTOR_1_Z` always?. As far as I see, at least in FluidApp, `ADJOINT_FLUID_VECTOR_1_Z` dof is only checked and used in 3D.

Reply:
Please see the discussion in #9467.